lib/jekyll/converters/markdown/kramdown_parser.rb

```diff
@@ -5,6 +5,11 @@ class KramdownParser
         def initialize(config)
           require 'kramdown'
           @config = config
+          # If kramdown supported highlighter enabled, use that
+          highlighter = @config['highlighter']
+          if highlighter == 'rouge' || highlighter == 'coderay'
```

Comment:
@parkr I’ve added this extra check so that we only set `['kramdown']['syntax_highlighter']` if rouge or coderay are set as the global highlighter. I’m a Ruby newbie, so please tell me if this works, or can be optimised! Thanks.